pkg/chains/storage/oci/legacy.go

```diff
@@ -280,8 +281,11 @@ func (b *Backend) RetrieveArtifact(ctx context.Context, obj objects.TektonObject
 
 func newDigest(cfg config.Config, imageName string) (name.Digest, error) {
 	// Override image name from config if set.
-	if r := cfg.Storage.OCI.Repository; r != "" {
-		imageName = r
+	if storageOCIRepository := cfg.Storage.OCI.Repository; storageOCIRepository != "" {
+		parts := strings.SplitN(imageName, "/", 3)
+		if len(parts) == 3 && (strings.ContainsRune(parts[0], '.') || strings.ContainsRune(parts[0], ':')) {
+			imageName = fmt.Sprintf("%s/%s", cfg.Storage.OCI.Repository, parts[2])
```

Comment:
This doesn't seem right. My understanding is that if `cfg.Storage.OCI.Repository` is set, then that's the full path to the repo where signatures and attestations will be pushed to. It is not meant to be a prefix to which a repo can be appended to.

Reply:
Yeah exactly I thought the same that it should be the full path of the repo where signatures and attestations should be pushed, but there are couple of things which are needed to push the signatures and attestations 

Consider you have given the value for `storage.oci.repository` as `docker.io/<username>/kaniko-chains` in the chains-config configMap

* First thing when it calls the [newDigest](https://github.com/tektoncd/chains/blob/main/pkg/chains/storage/oci/legacy.go#L281) function it returns an error which is from the [vendor](https://github.com/google/go-containerregistry/blob/main/pkg/name/digest.go#L67)  and hence the signing fails

* Now as you said 
> if cfg.Storage.OCI.Repository is set, then that's the full path to the repo where signatures and attestations will be pushed to

This is not true because consider you have set the value for `storage.oci.repository` as `docker.io/<username>/kaniko-chains` and artifacts are signed and pushed, but now you create a new taskrun which is different then it should not be pushed to the same artifact right 

Hence the value which should be set should be just the registry name and the repo name (considering if user doesn't provides registry name and just the repo name then as of today `docker.io` is used as by default)  

For example : `docker.io/<username>`